astroid/inference.py

```diff
@@ -100,14 +117,15 @@ def _infer_map(node, context):
             if not isinstance(double_starred, nodes.Dict):
                 raise exceptions.InferenceError(node=node,
                                                 context=context)
-            values.update(_infer_map(double_starred, context))
+            unpack_items = _infer_map(double_starred, context)
+            _update_with_replacement(values, unpack_items)
         else:
             key = helpers.safe_infer(name, context=context)
             value = helpers.safe_infer(value, context=context)
             if any(elem in (None, util.Uninferable) for elem in (key, value)):
                 raise exceptions.InferenceError(node=node,
                                                 context=context)
-            values[key] = value
+            _update_with_replacement(values, {key: value})
```

Comment:
I just tested that this pull request does not break the `duplicate-keys` in pylint since inference doesn't happen before the ast is analyzed. This pull request fixes a false positive in function `repeated-keywords`; If I did not update with replacement here, then the original issue would not be fixed.

Reply:
Alright, that's cool then!